test/lifecycle_test.dart

```diff
@@ -184,18 +184,48 @@ main() {
             'initialDerivedState': 'initial',
           };
 
-          final Map initialProps = unmodifiableMap({'getDerivedStateFromProps': (_, __, ___) => initialDerivedState});
+          const Map updatedDerivedState = const {
+            'initialDerivedState': 'updated',
+          };
+
+          final Map initialProps = {'getDerivedStateFromProps': (_, __, ___) => initialDerivedState};
```

Comment:
Could we make this and unmodifiable maps like they used to be? That reduces the risk of these maps changing unexpectedly and causing the wrong values to be checked in the expectations.